lib/pkg_editor/src/zlib.c

```diff
@@ -86,7 +86,12 @@ static void *zlib_load_library() {
   void *const library = (void *)LoadLibraryA(WINDOWS_ZLIB_PATH);
   return library;
 #else
-  return dlopen("libz.so", RTLD_NOW);
+  void *dlopen_handle = (void *)dlopen("libz.so.1", RTLD_NOW);
```

Comment:
[`dlopen(3)`](https://man7.org/linux/man-pages/man3/dlopen.3.html) returns a `void *` so you can omit the cast. The underlying idea is to only use casts when needed, and scrutinize the few casts that are indeed needed, especially C-style casts since these easily lead to undefined behaviour.